federatedscope/core/worker/client.py

```diff
@@ -314,6 +378,12 @@ def callback_funcs_for_join_in_info(self, message: Message):
                     num_sample = self._cfg.train.local_update_steps * \
                                  self.trainer.ctx.num_train_batch
                 join_in_info['num_sample'] = num_sample
+            elif requirement.lower() == 'client_info':
+                assert self.comm_bandwidth is not None and self.comp_speed \
+                       is not None, "The requirement join_in_info " \
+                                    "'client_info' does not exist."
+                join_in_info['client_info'] = self.model_size / \
```

Comment:
The key 'client_info' is much more general than what the corresponding value really is, right? BTW, I am wondering what if an FL algo exchanges information other than model parameters.

Reply:
Change `client_info` to `client_resource`.